### ISSUE TITLE
Simulating Different Acceleration Factors

### ISSUE THREAD

**Issue by srveale** [firasm/CEST#43]
To start off, here is what the signal history looks like for one frequency offset (0.2ppm), for acceleration factor of 8:
![signal history for accfac8](https://cloud.githubusercontent.com/assets/7737922/8141205/adb26354-1118-11e5-81d3-4644ed2c9f73.png)

And for acceleration factor of 24:
![signal history for accfac24](https://cloud.githubusercontent.com/assets/7737922/8141210/ba2f685c-1118-11e5-80fa-2a4a2b0431c1.png)

For brevity I won't post all of them, but I'll go straight to the point. The z-spectra at the centre of k-space:
![z-spectra for different accfacs](https://cloud.githubusercontent.com/assets/7737922/8141335/552ca094-111a-11e5-9419-3c14c0842b33.png)

Yes, the order is out of whack in places. AccFactor= 72 is above accFactor = 1 in signal magnitude, but has a smaller peak. My suspicion is that this is related to what was happening before with shorter saturation durations - notice the small scale oscillations. This was done using saturation duration =  500ms, and I'll try with 4000ms, but these take a long time because each frequency offset has as many as 72 saturation pulses.

Let me know if there is anything I should be looking at differently.

**Comment by DrSAR** [comment]
Can you state (just for posterity) saturation and repetition times

**Comment by DrSAR** [comment]
never mind - silly me.

**Comment by andrewcyung** [comment]
Could you plot the normalized spectra (i.e. divide by the baseline)?  It looks like the normalized spectra in terms of relative peak heights might look quite similar between acceleration factors.  But the absolute level of noise would be the same, so those spectra with a lower baseline (like acc=24) would have higher noise relative to the feature size.

**Comment by andrewcyung** [comment]
So do you think the fact that acc=72 has a higher saturation baseline than acc=1 is real?  I notice that there is not much difference between acc 24 and acc 36.  Maybe there are two competing effects in the accfactor dependence where one effect dominates at small accfactor, then both effects sort of cancel each other out at accfactor 24, then the other effect dominates all the way to accfactor 72.

**Comment by srveale** [comment]
I changed some parameters to be more in line with Kim's thesis. Dotted lines are because it ran out of colours.
![z-spectra for different acceleration factors 3 5ppm normalized](https://cloud.githubusercontent.com/assets/7737922/8149379/df7a9cc2-1275-11e5-8c1b-e99fe25d1a79.png)

I changed some parameters to be more in line with Kim's thesis:
T1water = 1.7s
T2water = 0.05s
T1species = 1.0s
T2species = 0.04s
exchange rate = 20/s

ti = 5 ms  - This is the time between saturation pulse and first flip. Possibly relevant to this discussion.
TR = 10ms   - This is the time between flip angles. 
Saturation duration = 500 ms, same as above.
Interfrequency delay = 7000ms

Acc=72 has a higher baseline because there is only one saturation pulse. Full saturation isn't reached during the 500ms pulse, and then signal is acquired right after. Compare this to a sequence that has more saturation pulses, not to mention several rotations that flip z-magnetization into the xy-plane and spoils it, before signal is acquired.

**Comment by srveale** [comment]
There is another effect at play. I had to write the next bit very slowly: With a high acceleration factor, there are many flips that go along with each acquisition, meaning that the z-magnetization actually reaches a *lower* point than it would have under the influence of saturation. The higher the accfactor, the more flips, the lower the magnetization gets. This also helps explain why higher accfactor leads to lower baseline, but I don't  know that this effect, by itself, would differentiate normalized peak size between accfactors. The next figure illustrates this effect.

![z-magnetization accfactor8](https://cloud.githubusercontent.com/assets/7737922/8149617/bd48e060-127e-11e5-8364-c5f3161ef3b2.png)

My gut tells me that the different peak sizes are a result of a combination of this 'aquisition-as-saturation' (better acronym pending) effect and a build-up of exchange-induced saturation over the course of many saturations. This is guesswork and maybe you guys would know better, but if there are many saturation pulses, there are many exchanges, and if T1 is slow enough these can build up.

To summarize the three effects that I notice:
-Accfac=72 does not reach saturation before the centre of k-space is acquired
-Many flips during acquisition means lower baseline
-More saturation pulses mean more exchange, possibly building up over time and producing bigger peaks.

I suspect that the different peaks heights will collapse to the same peak if saturation length is long enough. I'll investigate. I tried before but something broke during the looong simulation.

**Comment by srveale** [comment]
I ran it again for a saturation duration of 4000ms.

![z-spectra for different accfactors satdur4000 not normalized](https://cloud.githubusercontent.com/assets/7737922/8153055/92c23744-12db-11e5-8908-866cc04b61c0.png)


And normalized. The peaks are nearly the same now. 

![z-spectra for different accfactors satdur4000](https://cloud.githubusercontent.com/assets/7737922/8152993/33898e5e-12da-11e5-9bea-bcdf07806f91.png)

 I'll keep trying to interpret these results.

**Comment by srveale** [comment]
We wanted to look at one long acquisition with no acceleration vs the average of a few accelerated scans. I added some noise to each spectrum after they were generated. Here is the spectrum for accfactor =1, and the average of 5 spectra generated at accfactor = 8. Hopefully this is useful for deciding on how best to spend scanning time.

![acc factor 1 and an average of 5 accfactor 8 z-spectra](https://cloud.githubusercontent.com/assets/7737922/8312680/03a508ac-1993-11e5-9b49-d4c07f938e4a.png)

When the peak is as noticeable is this, it might not help to take the extra time obtaining spectra. But I think that for small peaks relative to noise, averaging over many acquisitions would be more useful.
